src/Kernel32/Kernel32.cs

```diff
@@ -853,6 +853,143 @@ public static partial class Kernel32
+        ///
+        /// This handle has the THREAD_ALL_ACCESS access right to the thread object.
+        ///
+        /// Windows Server 2003 and Windows XP:  This handle has the maximum access allowed by the security descriptor of the thread to the primary token of
+        /// the process.
+        ///
+        /// The function cannot be used by one thread to create a handle that can be used by other threads to refer to the first thread.The handle is always
+        /// interpreted as referring to the thread that is using it.A thread can create a "real" handle to itself that can be used by other threads, or
+        /// inherited by other processes, by specifying the pseudo handle as the source handle in a call to the DuplicateHandle function.
+        ///
+        /// The pseudo handle need not be closed when it is no longer needed.Calling the <see cref="CloseHandle(IntPtr)"/> function with this handle has no effect. If the pseudo
+        /// handle is duplicated by DuplicateHandle, the duplicate handle must be closed.
+        ///
+        /// Do not create a thread while impersonating a security context. The call will succeed, however the newly created thread will have reduced access
+        /// rights to itself when calling GetCurrentThread.The access rights granted this thread will be derived from the access rights the impersonated user
+        /// has to the process.Some access rights including THREAD_SET_THREAD_TOKEN and THREAD_GET_CONTEXT may not be present, leading to unexpected failures.
+        /// </remarks>
+        [DllImport(api_ms_win_core_processthreads_l1_1_1)]
+        public static extern SafeObjectHandle GetCurrentThread();
+
+        /// <summary>
+        /// Retrieves the boundaries of the stack that was allocated by the system for the current thread.
+        /// </summary>
+        /// <param name="LowLimit">A pointer variable that receives the lower boundary of the current thread stack.</param>
+        /// <param name="HighLimit">A pointer variable that receives the upper boundary of the current thread stack.</param>
+        /// <remarks>
+        /// It is possible for user-mode code to execute in stack memory that is outside the region allocated by the system when the thread was created.
+        /// Callers can use the GetCurrentThreadStackLimits function to verify that the current stack pointer is within the returned limits.
+        ///
+        /// To compile an application (C, C++) that uses this function, set _WIN32_WINNT >= 0x0602. This corresponds to a min. platform
+        /// requirement of Windows 8/Windows Server 2012 for using this function.
+        /// </remarks>
+        [DllImport(api_ms_win_core_processthreads_l1_1_1)]
+        public static unsafe extern void GetCurrentThreadStackLimits(
+            [Friendly(FriendlyFlags.Out)] ulong* LowLimit,
+            [Friendly(FriendlyFlags.Out)] ulong* HighLimit);
```

Comment:
These are `PULONG_PTR` in native headers. Shouldn't they be `UIntPtr` here then so they are 32-bit on x86?
Also as mentioned elsewhere, `out` instead of pointers is adequate IMO for primitive types.

Reply:
You're right - I'll update.